library/std/src/sys/windows/thread.rs

```diff
@@ -101,11 +101,11 @@ impl Thread {
 
 pub fn available_parallelism() -> io::Result<NonZeroUsize> {
     let res = unsafe {
-        let mut sysinfo: c::SYSTEM_INFO = crate::mem::zeroed();
-        c::GetSystemInfo(&mut sysinfo);
-        sysinfo.dwNumberOfProcessors as usize
+        // FIXME: windows::Win32::System::SystemServices::ALL_PROCESSOR_GROUPS should be u16, not u32
+        c::GetActiveProcessorCount(65535u16) as usize
```

Comment:
Would you be able to file an [upstream](https://github.com/microsoft/win32metadata) bug for that fixme? Also

```suggestion
        // FIXME: windows::Win32::System::SystemServices::ALL_PROCESSOR_GROUPS should be u16, not u32
        c::GetActiveProcessorCount(c::ALL_PROCESSOR_GROUPS as u16) as usize
```
I'd prefer to still use the `ALL_PROCESSOR_GROUPS` constant here regardless. I think `as` is fine in this case but `try_into().unwrap()` is perhaps a safer option.

Reply:
Already here https://github.com/microsoft/windows-rs/issues/2608

Reply:
Great!

Reply:
If definition will be fixed in near feature, i will replace it.